src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -543,6 +544,62 @@ static bool simplify_reduction(shared_ptr<Node> n)
     return true;
 }
 
+static bool replace_transpose_with_reshape(shared_ptr<Node> n)
+{
+    bool rc = false;
+    auto transpose = as_type_ptr<op::Transpose>(n);
+
+    if (transpose)
+    {
+        auto perm = as_type_ptr<op::Constant>(n->get_argument(1));
+        auto perm_value = perm->get_vector<int64_t>();
+        auto data = n->get_argument(0);
```

Comment:
This should be `n->input_value(0)` and then use `get_partial_shape()` on that. You need to check that the partial shape is static before calling `to_shape()` or you'll get an error.